waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/model/FederatedMetaStore.java

```diff
@@ -67,6 +70,10 @@ public String getDatabasePrefix() {
   }
 
   public void setMappedDatabases(List<String> mappedDatabases) {
+    if (mappedDatabases == null) {
+      ValidationError validationError = ValidationError.builder().error("mapped-databases should not be null").build();
```

Comment:
This error message doesn't show up in the error. Waggle Dance just shows the type of error, so maybe I'm doing something wrong?

Reply:
What happens if you do this?

`public void setMappedDatabases(@NotNull List<String> mappedDatabases)`

Instead of doing the logic ourselves? I see we do something like that in `com.hotels.bdp.waggledance.api.federation.service.FederationService` (although I'm not sure exactly how that works since `FederationService` is an interface). I'm also not sure what enforces the checks and how the exceptions are thrown but the base `AbstractMetaStore` has similar annotations on the member variables (e.g. `accessControlType`) so I assume it "just works". Another option would be to instead do it on line 25 when we declare `mappedDatabases` instead of on the method parameter here.

Reply:
For these, this test passes successfully 😞 
```java
@Test
public void nullMappedDatabases() {
  metaStore.setMappedDatabases(null);
  List<String> mappedDatabases = metaStore.getMappedDatabases();
  assertThat(mappedDatabases, is(nullValue()));
}
```
but I'll see what happens on a real environment

Reply:
Actually, I need to check the violations, so you can ignore my previous comment 🙂

Reply:
Putting `@NotNull` to the function parameter doesn't change anything, but putting it to the class variable makes WD gives an instant validation error and it also
```
2019-02-13 15:34:45.066 ERROR 24262 --- [           main] c.h.b.w.WaggleDance                      : Validation errors:
2019-02-13 15:34:45.066 ERROR 24262 --- [           main] c.h.b.w.WaggleDance                      : ConstraintViolationImpl{interpolatedMessage='must not be null', propertyPath=federatedMetaStores[0].mappedDatabases, rootBeanClass=class com.hotels.bdp.waggledance.api.model.Federations, messageTemplate='{javax.validation.constraints.NotNull.message}'}
```

So I'll change this to have `private @NotNull List<String> mappedDatabases = Collections.emptyList();`